### ISSUE TITLE
`jit.trace` fails to capture single-element tensor passed into op requiring scalar-type argument

### ISSUE THREAD

**Issue by gmagogsfm** [pytorch/pytorch#57592]
## 🐛 Bug

```
class M(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.p = torch.tensor(1.0)

    def forward(self, x):
        return torch.dropout(x, self.p, False) 

m = M()
traced = torch.jit.trace(m, torch.rand(2, 3))
print(traced.graph)
```

Output:
```
graph(%self : __torch__.M,
      %x : Float(2, 3, strides=[3, 1], requires_grad=0, device=cpu)):
  %8 : float = prim::Constant[value=1.]() # simple_trace_const.py:9:0
  %9 : bool = prim::Constant[value=0]() # simple_trace_const.py:9:0
  %10 : Float(2, 3, strides=[3, 1], requires_grad=0, device=cpu) = aten::dropout(%x, %8, %9) # simple_trace_const.py:9:0
  return (%10)
```

We can see that second argument to `aten::dropout` is baked in as a constant float when it should have been `self.p`.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
